src/classes/SftpSessionHandler.ts

```diff
@@ -0,0 +1,388 @@
+import type {
+  Attributes,
+  FileEntry,
+  SFTPWrapper,
+} from 'ssh2';
+import type { Archive } from '@permanentorg/sdk';
+
+const getFileType = (filePath: string): number => {
+  // TODO: this is not a real rule -- directories can have extensions and files don't have to have them.
+  if (filePath.includes('.')) {
+    return fs.constants.S_IFREG;
+  }
+  return fs.constants.S_IFDIR;
+};
+
+const SFTP_STATUS_CODE = ssh2.utils.sftp.STATUS_CODE;
+
+const isDirectoryMode = (mode: number): boolean => (mode & fs.constants.S_IFMT) == fs.constants.S_IFDIR;
+
+const generateDefaultMode = (baseType: number): number => (
+  baseType
+  | fs.constants.S_IRWXU // Read, write, execute for user
+  | fs.constants.S_IRWXG // Read, write, execute for group
+  | fs.constants.S_IRWXO // Read, write, execute for other
+);
+
+const generateDefaultAttributes = (fileType: number): Attributes => ({
+  mode: generateDefaultMode(fileType),
+  uid: 0,
+  gid: 0,
+  size: 0,
+  atime: 0,
+  mtime: 0,
+});
+
+const getFilename = (filePath: string): string => {
+  return filePath;
+};
+
+const getLongname = (
+  filename: string,
+  attributes: Attributes,
+  owner = 'nobody',
+  group = 'nogroup',
+): string => {
+  const directoryFlag = isDirectoryMode(attributes.mode) ? 'd' : '-';
+  return `${directoryFlag}rwxrwxrwx 1 ${owner} ${group} 3 Dec 8 2009 ${filename}`;
+};
+
+const generateFileEntry = (
+  filePath: string,
+  attributes: Attributes,
+): FileEntry => ({
+  filename: filePath,
+  longname: getLongname(
+    getFilename(filePath),
+    attributes,
+  ),
+  attrs: attributes,
+});
+
+const generateHandle = (): string => uuidv4();
+
+const loadArchives = async (authToken: string): Promise<FileEntry[]> => {
+  const archives = await getArchives({
+    bearerToken: authToken,
+  });
+  return archives.map((archive: Archive) => {
+    const archiveDirectoryPath = `/archives/${archive.name} (${archive.id})`;
+    return generateFileEntry(
+      archiveDirectoryPath,
+      generateDefaultAttributes(fs.constants.S_IFDIR),
+    );
+  });
+};
+
+const loadArchive = async (authToken: string, archiveName: string): Promise<FileEntry[]> => {
+  return [
+    generateFileEntry(
+      `${archiveName}/MyFiles`,
+      generateDefaultAttributes(fs.constants.S_IFDIR),
+    ),
+    generateFileEntry(
+      `${archiveName}/Shares`,
+      generateDefaultAttributes(fs.constants.S_IFDIR),
+    ),
+    generateFileEntry(
+      `${archiveName}/Public`,
+      generateDefaultAttributes(fs.constants.S_IFDIR),
+    ),
+    generateFileEntry(
+      `${archiveName}/Apps`,
+      generateDefaultAttributes(fs.constants.S_IFDIR),
+    ),
+  ];
+};
+
+const loadPath = async (requestedPath: string, authToken: string): Promise<FileEntry[]> => {
+  if (requestedPath === '/') {
+    return [
+      generateFileEntry(
+        '/archives',
+        generateDefaultAttributes(fs.constants.S_IFDIR),
+      ),
+    ];
+  } else if (requestedPath === '/archives') {
```

Comment:
Maybe these should go in a named function so it's clear something like `isArchiveRootPath`